src/ExchangeRateProviders/ExchangeRateHostProvider.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\Exchange\ExchangeRateProviders;
+
+use Illuminate\Http\Client\Factory;
+use Worksome\Exchange\Contracts\ExchangeRateProvider;
+use Worksome\Exchange\Support\Rates;
+
+final class ExchangeRateHostProvider implements ExchangeRateProvider
+{
+    private FixerProvider $fixerProvider;
+
+    public function __construct(private Factory $client)
+    {
+        $this->fixerProvider = new FixerProvider($this->client, '', 'https://api.exchangerate.host');
```

Comment:
If we get a third implementation that has the same API, then let's move the current fixer into an abstract class instead, but for now let's just do it with a decorator